src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -285,16 +287,28 @@ void LinearIR::unregister_expression(const ExpressionPtr& expr) {
         input->remove_consumer(expr->get_input_port(i));
     }
 
-    m_node2expression_map.erase(expr->get_node());
+    const auto& node = expr->get_node();
```

Comment:
BTW I see that before we have opportunity to erase parameter or result op - but it should be forbidden. Should we implement the same way here as for `register_expression`? Don't allow to unregister parameter or result if it's not `manual_build` ? What do you think?

Reply:
Good idea I think. Moreover, we can probably prohibit parameter/result unregistration at all -- this functionality is not needed even in tests

Reply:
Done, thanks